src/Classes/Canvas.ts

```diff
@@ -154,7 +158,159 @@ export class Canvas {
+        }
+
+        let colorLine: IRgb | undefined;
+
+        /*
+         * particles connecting line color:
+         *
+         *  random: in blink mode : in every frame refresh the color would change
+         *          hence resulting blinking of lines
+         *  mid: in consent mode: sample particles color and get a mid level color
+         *                        from those two for the connecting line color
+         */
+
+        if (container.particles.lineLinkedColor === "random") {
+            colorLine = Utils.getRandomColorRGBA();
+        } else if (container.particles.lineLinkedColor == "mid" && p1.color && p2.color) {
+            const sourceColor = p1.color;
+            const destColor = p2.color;
+
+            colorLine = {
+                b: Math.floor(Utils.mixComponents(sourceColor.b, destColor.b, p1.radius, p2.radius)),
+                g: Math.floor(Utils.mixComponents(sourceColor.g, destColor.g, p1.radius, p2.radius)),
+                r: Math.floor(Utils.mixComponents(sourceColor.r, destColor.r, p1.radius, p2.radius)),
+            };
+        } else {
+            colorLine = container.particles.lineLinkedColor as IRgb;
+        }
+
+        const width = container.retina.lineLinkedWidth;
+
+        CanvasUtils.drawLineLinked(ctx, width, pos1, pos2, options.backgroundMask.enable, colorLine, opacity);
+    }
+
+    public drawConnectLine(p1: Particle, p2: Particle): void {
+        const lineStyle = this.lineStyle(p1, p2);
+
+        if (!lineStyle) {
+            return;
+        }
+
+        const ctx = this.context;
+
+        if (!ctx) {
+            return;
+        }
+
+        CanvasUtils.drawConnectLine(ctx, this.container.retina.lineLinkedWidth, lineStyle, p1.position, p2.position);
+    }
+
+    public drawGrabLine(particle: Particle, opacity: number, mousePos: ICoordinates): void {
+        const container = this.container;
+        const options = container.options;
+        const optColor = options.particles.lineLinked.color;
+
+        let lineColor = container.particles.lineLinkedColor || Utils.hexToRgb(optColor);
+
+        if (lineColor == "random") {
+            lineColor = Utils.getRandomColorRGBA();
+        }
+
+        container.particles.lineLinkedColor = lineColor;
+
+        let colorLine: IRgb = {r: 127, g: 127, b: 127};
+        const ctx = container.canvas.context;
+
+        if (!ctx) {
+            return;
+        }
+
+        if (container.particles.lineLinkedColor == "random") {
+            colorLine = Utils.getRandomColorRGBA();
+        } else {
+            colorLine = container.particles.lineLinkedColor as IRgb || colorLine;
+        }
+
+        const beginPos = {
+            x: particle.position.x + particle.offset.x,
+            y: particle.position.y + particle.offset.y,
+        };
+
+        CanvasUtils.drawGrabLine(ctx, container.retina.lineLinkedWidth, beginPos, mousePos, colorLine, opacity);
+    }
+
+    public drawParticle(particle: Particle): void {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Declaration of public instance method not allowed after declaration of private instance method. Instead, this should come after constructors.](https://app.codacy.com/manual/ar3s/tsparticles/pullRequest?prid=5171185)